WordPress/Classes/Services/UserService.swift

```diff
@@ -3,62 +3,94 @@ import Combine
 import WordPressAPI
 import WordPressUI
 
+protocol ObjectCache<Cacheable>: Actor {
+
+    associatedtype Cacheable: (Identifiable & Sendable)
+
+    func get(_ id: Cacheable.ID) -> Cacheable?
+    func store(_ object: Cacheable)
+    func store(_ objects: [Cacheable])
+    func all() -> [Cacheable]
+    func invalidate(_ id: Cacheable.ID)
+}
+
+actor InMemoryCache<Cacheable: Identifiable & Sendable>: ObjectCache {
+
+    private var objects: [Cacheable.ID: Cacheable] = [:]
+
+    func get(_ id: Cacheable.ID) -> Cacheable? {
+        objects[id]
+    }
+
+    func store(_ object: Cacheable) {
+        objects[object.id] = object
+    }
+
+    func store(_ newObjects: [Cacheable]) {
+        for object in newObjects {
+            objects[object.id]  = object
+        }
+    }
+
+    func all() -> [Cacheable] {
+        [Cacheable](objects.values)
+    }
+
+    func invalidate(_ id: Cacheable.ID) {
+        self.objects.removeValue(forKey: id)
+    }
+}
+
 /// UserService is responsible for fetching user acounts via the .org REST API – it's the replacement for `UsersService` (the XMLRPC-based approach)
 ///
 actor UserService: UserServiceProtocol {
     private let client: WordPressClient
 
-    private var fetchUsersTask: Task<[DisplayUser], Error>?
-
-    private(set) var users: [DisplayUser]? {
-        didSet {
-            if let users {
-                usersUpdatesContinuation.yield(users)
-            }
-        }
-    }
-    nonisolated let usersUpdates: AsyncStream<[DisplayUser]>
-    private nonisolated let usersUpdatesContinuation: AsyncStream<[DisplayUser]>.Continuation
+    private let cache: any ObjectCache<DisplayUser>
 
     private var currentUser: UserWithEditContext?
 
-    init(client: WordPressClient) {
+    private var tasks: [Task<(), any Error>] = []
+
+    init(client: WordPressClient, cache: any ObjectCache<DisplayUser>) {
         self.client = client
-        (usersUpdates, usersUpdatesContinuation) = AsyncStream<[DisplayUser]>.makeStream()
+        self.cache = cache
     }
 
     deinit {
-        usersUpdatesContinuation.finish()
-        fetchUsersTask?.cancel()
+        for task in tasks {
+            task.cancel()
+        }
     }
 
-    func fetchUsers() async throws -> [DisplayUser] {
-        let users = try await createFetchUsersTaskIfNeeded().value
-        self.users = users
-        return users
-    }
+    func fetchPaginatedUsers() -> AsyncThrowingStream<[DisplayUser], Error> {
```

Comment:
If I understand this correclty, this stream's outputs are cached users (which may be empty or all the pre-fetched users), users at page 1, users at page 2, ..., END. If that's the case, how `fetchPaginatedUsers` is used in the View Model is wrong. The View Model should not collect all the stream's outputs into an array.